app/models/proposal.rb

```diff
@@ -270,19 +264,22 @@ def create_new_observation(user, adder, reason)
     # invalidate relation cache so we reload on next access
     self.observers(true)
     # when explicitly adding an observer using the form in the Proposal page...
-    if adder
-      add_observation_comment(user, adder, reason) unless reason.blank?
+    if adder && reason
+      add_observation_comment(user, adder, reason)
```

Comment:
oh! the next line isn't a change in behavior - this mailer (`Dispatcher.on_observer_added(observation, reason)`) was formerly called in  callback on `comment#create`, but I removed that callback to give us more control over behavior and make it easier to suppress comment emails in certain situations, like when a proposal is cancelled. 

I did try to add specs for each situation where we create a comment and send a comment email.